### PR TITLE
Make new SSO work with pre-existing shib acounts

### PR DESCRIPTION
**Pull request by kluo**
@stvstnfrd

### CODE REVIEW

File: common/djangoapps/third_party_auth/pipeline.py [hunk 1]
```diff
@@ -533,7 +534,7 @@ def should_force_account_creation():
             # account corresponds to them yet, if any.
             if should_force_account_creation():
                 return dispatch_to_register()
-            return dispatch_to_login()
+            login_user(strategy.request)
```
Comment (caesar2164) [inline common/djangoapps/third_party_auth/pipeline.py]: couldn't this just be `student.views.login_user(strategy.request)` and then you could remove the new import on line 82?

**Comment by stvstnfrd** [comment]
@kluo Tests are failing; can you investigate?

File: common/djangoapps/third_party_auth/pipeline.py (hunk 1 above)
Comment (stvstnfrd) [inline common/djangoapps/third_party_auth/pipeline.py]: In fact, this import is blowing up the tests:

```
    common/djangoapps/third_party_auth/pipeline.py line 82 in <module>
      from student.views import login_user
   ImportError: cannot import name login_user
```

**Comment by stvstnfrd** [comment]
@kluo We'll also need unit tests.

**Comment by stvstnfrd** [comment]
and this will definitely need rebased over the new master.

**Comment by stvstnfrd** [comment]
@kluo I'm killing this now.

**Pull request closed without merging**